### DISCUSSION TITLE
Pandoc syntax conversion issue

### DISCUSSION THREAD

**Discussion by wafflesatdawn** [quarto-dev/quarto-cli#5428] [Q&A] [answered]
I'm using Quarto with Hugo and a theme that incorporates a very odd Katex syntax for math equations: it users \\( and \\) as delimiters. 

Rendering pages from Quarto causes the parentheses to be stripped and several lines of js/jquery to appear at the start of the .md file. To get my code to display correctly, I need to delete the js/jquery and correct the delimiters. I've tested on other pages and they don't have similar issues with the random js.

I think it's a Pandoc setting that's causing the behavior with the delimiters but have no idea about the js. If there could be a toggle or if Katex could be supported that'd be appreciated.

**Comment by mcanouil** [comment]
Could you share some small example for us to reproduce?
Note that I believe the issue you are having is as you said with Pandoc and not with Quarto directly.
Or, there might be some changes to be made in your katex setup (https://katex.org/).

**Reply by wafflesatdawn** [reply]
Reproducing the issue might be a bit involved, but assuming you have Hugo already installed, to install the theme:

`cd mywebsite
git init
git submodule add -b main https://github.com/nunocoracao/blowfish.git themes/blowfish
hugo new .\content\posts\collaborative-filtering\new.md
`
Then for the new file:

````
---
title: "New"
date: 2023-05-04T19:29:52-07:00
format: hugo-md
jupyter: python3
draft: true
---
```{python}
from fastai.collab import *
from fastai.tabular.all import *
from pathlib import Path
# import zipfile

gamepath = Path('./data/games.csv')
games = pd.DataFrame([range(1,5), range(1,5)])
games.head()
```
{{< katex >}}
$$ax^2+bx+c$$

inline {{< katex >}}\\(ax^2+bx+c\\) inline
````

Then rendering the .qmd to .md:

`quarto render .\content\posts\collaborative-filtering\new.qmd`

Expected output:

````
---
title: "New"
date: 2023-05-04T19:29:52-07:00
format: hugo-md
jupyter: python3
draft: true
---

<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


``` python
from fastai.collab import *
from fastai.tabular.all import *
from pathlib import Path
# import zipfile

gamepath = Path('./data/games.csv')
games = pd.DataFrame([range(1,5), range(1,5)])
games.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>

|     | 0   | 1   | 2   | 3   |
|-----|-----|-----|-----|-----|
| 0   | 1   | 2   | 3   | 4   |
| 1   | 1   | 2   | 3   | 4   |

</div>

{{< katex >}}

$$ax^2+bx+c$$

inline {{< katex >}}\\ax^2+bx+c\\ inline
````

The js will appear at the top and the delimiters for inline notation will  be modified. Not sure about how to modify the katex setup... It might be something I can live with. As for Pandoc, isn't it part of Quarto's render process or something?

**Reply by cscheid** [reply]
Can you try to format your post better? If your file has triple ticks, surround it with quadruple ticks, and so on.

**Reply by mcanouil** [reply]
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us> regarding how to include Quarto document in a post.

PS: why are you using the double dollars for multiline equations (display) and not the single dollar syntax for inline?

**Reply by wafflesatdawn** [reply]
I've edited my reply using the 4x backticks as suggested--hope that helps!

Re: $ and $$ syntax, my Hugo theme won't process them the same way Quarto does. $$ syntax is allowed for separate lines but there is no single dollar version. Instead inline has to use `\\(` and `))\`.

**Reply by mcanouil** [reply, marked as answer]
Unfortunately, your example is not reproducible.

Note that to have the correct Hugo syntax, you need for some reasons to double escape the equation.

````qmd
---
title: "New"
date: 2023-05-04T19:29:52-07:00
format: hugo-md
---

{{< katex >}}

$$ax^2+bx+c$$

inline {{< katex >}}\\\\(ax^2+bx+c\\\\) inline
````

Or include it in a raw markdown block[^1]

````qmd
---
title: "New"
date: 2023-05-04T19:29:52-07:00
format: hugo-md
---

```{=markdown}
{{< katex >}}

$$ax^2+bx+c$$

inline {{< katex >}}\\(ax^2+bx+c\\) inline
```
````

or raw inline markdown[^1]

````qmd
---
title: "New"
date: 2023-05-04T19:29:52-07:00
format: hugo-md
---

{{< katex >}}

$$ax^2+bx+c$$

inline `{{< katex >}}\\(ax^2+bx+c\\)`{=markdown} inline
````

[^1]: See <https://quarto.org/docs/output-formats/hugo.html#shortcodes>.

**Reply by wafflesatdawn** [reply]
Thanks for the reference link! The notation is now displaying correctly after escaping to raw markdown.
